packages/gatsby-remark-images/src/index.js

```diff
@@ -30,11 +31,18 @@ module.exports = (
   const options = _.defaults(pluginOptions, defaults)
 
   // This will only work for markdown syntax image tags
-  const markdownImageNodes = select(markdownAST, `image`)
+  let markdownImageNodes = []
 
   // This will also allow the use of html image tags
   const rawHtmlNodes = select(markdownAST, `html`)
 
+  visitWithParents(markdownAST, `image`, (node, ancestors) => {
+    const inLink = ancestors.some(ancestor => ancestor.type === `link`)
+    if (!inLink) {
+      markdownImageNodes.push(node)
+    }
```

Comment:
This is broken. We still need to fix the relative path, which I'm pretty sure @pieh told me to do and I just... didn't.

Reply:
I think we need to process images in both cases (when it's inside link and when it's not) - difference should be whether we wrap image in link here - https://github.com/jlengstorf/gatsby/blob/af209f9215d8cd9c5ec2f3fda1cbdae1ec40027d/packages/gatsby-remark-images/src/index.js#L175-L188